scalding-core/src/test/scala/com/twitter/scalding/FileSourceTest.scala

```diff
@@ -216,13 +216,8 @@ class FileSourceTest extends WordSpec with Matchers {
       pathIsGood("test_data/2013/04/") shouldBe false
     }
 
-    "reject an empty directory" in {
-      pathIsGood("test_data/2013/05/") shouldBe false
-      pathIsGood("test_data/2013/05/*") shouldBe false
-    }
-
-    "reject a directory with only _SUCCESS when specified as a glob" in {
-      pathIsGood("test_data/2013/06/*") shouldBe false
+    "accept a directory with only _SUCCESS when specified as a glob" in {
```

Comment:
I agree this is a better contract. Basically, `_SUCCESS` should imply the path is good. This whole jank was about cases where there was no `_SUCCESS` and it was a crazy attempt to at least warn on totally missing data (mostly for people that had job schedulers that didn't know about dependencies).